File: Week4/Saturday/bubble-sort/python/bubble_sort.py
sequence = [4, 3, 5, 0, 1]


def bubble_sort(seq):
    idx = 1
    swaps = 0
    while idx < len(seq):
        previous = seq[idx - 1]
        current = seq[idx]
        if previous > current:
            hold = previous
            seq[idx], seq[idx - 1] = hold, current
            idx = 1
            swaps += 1
        else:
            idx += 1
    print(f"Final result: {seq}")
    print(f"Swaps: {swaps}")
    return seq


print(bubble_sort(sequence))

File: Week4/Saturday/bubble-sort/js/bubbleSort.js
var sequence = [4, 3, 5, 0, 1]


function bubbleSort(arr) {
    let index = 1;
    let swaps = 0;
    while (index < arr.length) {
        let previous = arr[index-1]   
        let current = arr[index]
        if (previous > current) {
            let hold = previous
            arr[index-1] = current 
            arr[index] = hold
            index = 1
            swaps++ 
        } else {
            index++
        }
    }
    console.log("Final result: " + arr)
    console.log("Swaps: " + swaps)
    return arr
}


console.log(bubbleSort(sequence))